plugin/src/main/kotlin/org/gradleweaver/plugins/jlink/JLinkPlugin.kt

```diff
@@ -10,21 +14,72 @@ open class JLinkPlugin : Plugin<Project> {
         const val EXTENSION_NAME = "jlink"
         const val JLINK_TASK_GROUP = "JLink"
         const val JLINK_TASK_NAME = "jlinkGenerate"
-        const val JLINK_ZIP_TASK_NAME = "jlinkZip"
+        const val JLINK_ARCHIVE_TASK_NAME = "jlinkArchive"
+
+        private fun capitalizeAndJoinWords(words: String): String {
+            return words.split(' ').joinToString(separator = "") { it.capitalize() }
+        }
+
+        fun generateJLinkTaskName(configurationName: String): String {
+            return "$JLINK_TASK_NAME${capitalizeAndJoinWords(configurationName)}"
+        }
+
+        fun generateJLinkArchiveTaskName(configurationName: String): String {
+            return "$JLINK_ARCHIVE_TASK_NAME${capitalizeAndJoinWords(configurationName)}"
+        }
     }
 
     override fun apply(project: Project) {
         val extension = project.extensions.create(EXTENSION_NAME, JLinkExtension::class, project)
 
-        val jlinkTask = project.tasks.register(JLINK_TASK_NAME) {
-            group = JLINK_TASK_GROUP
-            description = "Generates a native Java runtime image"
+        extension.configure.all {
+            generateTasks(this, project)
         }
+    }
 
-        val jlinkZipTask = project.tasks.register(JLINK_ZIP_TASK_NAME) {
-            group = JLINK_ZIP_TASK_NAME
-            description = "Generates a ZIP file of a native Java runtime image"
+    private fun generateTasks(options: JLinkOptions, project: Project) {
+        val jlinkTaskName = generateJLinkTaskName(options.name)
+        val archiveTaskName = generateJLinkArchiveTaskName(options.name)
+        project.tasks.register(jlinkTaskName, JLinkTask::class.java).configure {
+            group = JLINK_TASK_GROUP
+            description = "Generates a native Java runtime image for '${options.name}'."
+            copyFromOptions(options)
         }
+        if (OperatingSystem.current().isWindows) {
+            project.tasks.register(archiveTaskName, Zip::class.java) {
+                group = JLINK_TASK_GROUP
+                description = "Generates a .zip archive file of a native Java runtime image for '${options.name}'."
+                from(project.tasks.getByName(jlinkTaskName).outputs)
+            }
+        } else if (OperatingSystem.current().isUnix) {
+            project.tasks.register(archiveTaskName, Tar::class.java) {
+                group = JLINK_TASK_GROUP
+                description = "Generates a .tar.gz archive file of a native Java runtime image for '${options.name}'."
+                from(project.tasks.getByName(jlinkTaskName).outputs)
+                compression = Compression.GZIP
+                extension = "tar.gz"
+            }
+        }
```

Comment:
Why generate a different artifact per OS? Why not just generate both? Also, why default to GZIP? The user can configure that easily if they want to.